### PR TITLE
[draft] migrate automation to new build cli

### PR DESCRIPTION
**Pull request by sonalivdeshpande**

### CODE REVIEW

File: build-tools/packages/build-cli/src/commands/merge.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+        const resolvedRoot = await getResolvedFluidRoot();
+        const gitRepo = new GitRepo(resolvedRoot);
+
+        // check if PR exists
+        if (await prExists(flags.githubToken, "Automation: Main Next Integrate")) {
+            this.exit(-1);
+        }
+
+        const lastMergedCommit = await gitRepo.mergeBase(flags.sourceBranch, flags.targetBranch);
```
Comment (sonalivdeshpande) [inline build-tools/packages/build-cli/src/commands/merge.ts:144]: makes this as a function to check main/next state

File: build-tools/packages/build-cli/src/commands/merge.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import { BaseCommand } from "../base";
+
+const owner = "microsoft";
+const repo = "FluidFramework";
+
+async function prExists(token: string, title: string): Promise<boolean> {
+    const octokit = new Octokit({ auth: token });
+    const response = await octokit.request("GET /repos/{owner}/{repo}/pulls", { owner, repo });
+
+    for (const data of response.data) {
```
Comment (RishhiB) [inline build-tools/packages/build-cli/src/commands/merge.ts:17]: Just a small nit, might not even be nessesary but thought of a cool one liner for lines 17-23; 
`return response.data.some(data => data.title === title)`

File: build-tools/package.json [hunk 3]
```diff
@@ -85,6 +85,9 @@
     ],
     "temp-directory": "nyc/.nyc_output"
   },
+  "dependencies": {
+    "@octokit/core": "^4.0.4"
+  },
```
Comment (tylerbutler) [inline build-tools/package.json:90]: Is this needed?

Comment (sonalivdeshpande) [reply build-tools/package.json:90]: this is not needed...

**Pull request closed without merging**